src/Oscoin/API/HTTP/Client.hs

```diff
@@ -53,14 +57,20 @@ newtype HttpClientT m a = HttpClientT (ReaderT (Requester m) m a)
 instance MonadTrans HttpClientT where
     lift ma = HttpClientT $ lift ma
 
-instance (Monad m, MonadIO m) => MonadClient (HttpClientT m) where
+instance ( Buildable (Hash c)
+         , Serialise (BlockHash c)
+         , Serialise (PK c)
+         , Serialise (Signature c)
```

Comment:
Can we not say in one place: "given a `Serialise c`, `PK c` is also a `Serialise`? And likewise for all the others?

Reply:
I'm seeing this pattern repeated in a bunch of places is why I ask, and would be nice if we just needed something like `Serialise c`

Reply:
This was a deliberate design decision from my side, as yes, it leads to very big constraints, but they are all unrolled in one place and not hidden behind a type alias which then contains dozens of other instances. Plus, being so fine grained means we are paying only from what we need -- If you look at the `Oscoin.Test.Crypto` module you will see I did something similar by creating a kitchen sink `IsCrypto` alias.

While we could potentially do something similar, we are now making most of our functions needlessly constrained also on stuff they don't directly use. If we start breaking down each individual functionalities into smaller type alias we eventually bundle all them together, we are back into those crazy layer of indirections where you have to expand 10 different type synonyms to get the info you want.

At the end of the day I think it comes down to personal taste, and I am happy to stick with what the majority thinks 😉

Reply:
I wasn't suggesting a type alias, but rather a way to tell the compiler that given a `Serialise c`, and for ex. `instance Serialise c => Serialise (Block c)`, we wouldn't need `Serialise (Block c)` everywhere. Maybe it can't be done?

Reply:
Ah, sorry, I did got the impression I was answering the wrong question while typing it 😉 

Alas it can't be done, and if you scroll through the code you will see I have done this as-aggressively-as-possible for instances which could actually be written that way.

In this particular example, if we take a look at:

``` haskell
Serialise (BlockHash c), Serialise (PK c), Serialise (Signature c)
```
The blocker here is that both `BlockHash` is just an alias for `Hash`, and the latter, `PK` and `Signature` are _all_ type families, which makes simply not possible (at least not in a way that I know about!) to write such typeclass constraints the way you want 😢 .

This is one drawback of the current approach I am afraid 😞 you gain power, but you also gain fatter constraints. There is no free lunch alas!